### ISSUE TITLE
Invalid version format (Test::Deep)

### ISSUE THREAD

**Issue by wollmers** [perl11/cperl#51]
Installed with perlbrew on OSX 10.9.5

```
$ wget https://github.com/perl11/cperl/archive/cperl-5.22.1.tar.gz \
    -O cperl-cperl-5.22.1.tar.gz
$ perlbrew install --as=cperl-5.22.1 $(pwd)/cperl-cperl-5.22.1.tar.gz
```

```
helmut@helmut:~/github/LCS-XS$ cpanm .
--> Working on .
Configuring /Users/helmut/github/LCS-XS ... OK
==> Found dependencies: Test::Deep, LCS
--> Working on Test::Deep
Fetching http://www.cpan.org/authors/id/R/RJ/RJBS/Test-Deep-0.117.tar.gz ... OK
Configuring Test-Deep-0.117 ... OK
Invalid version format (non-numeric data) at /Users/helmut/perl5/perlbrew/perls/cperl-5.22.1/lib/5.22.1/Safe.pm line 51.
BEGIN failed--compilation aborted at /Users/helmut/perl5/perlbrew/perls/cperl-5.22.1/lib/5.22.1/Safe.pm line 51.
Compilation failed in require at /Users/helmut/perl5/perlbrew/bin/cpanm line 382.
BEGIN failed--compilation aborted at /Users/helmut/perl5/perlbrew/bin/cpanm line 382.
Compilation failed in require at /Users/helmut/perl5/perlbrew/bin/cpanm line 36.
```

**Comment by wollmers** [comment]
It's a problem with cpanm or one of the dependencies of cpanm. Can't narrow the problem quickly.

This works:

```
cpan Test::Deep
```

**Comment by rurban** [comment]
Yes, that's a bad chicken or egg problem. 
I wanted to add 'c' version markers to mark cperl changed modules somehow, but only cperl or patched toolchain modules can understand those 'c' suffices. When bootstrapping you will trip over it.

**Comment by rurban** [comment]
Note that I have to keep something like the 'c' suffix for typed variants of the modules in signatures. 
perl5 does not even want to parse those types.

**Comment by wollmers** [comment]
In the case of cpanm we can modify cperl-5.22.1/lib/5.22.1/Safe.pm as an ugly workaround:

```
# XXX hack for cperl
# XXX use Opcode 1.01, qw( # XXX old code
use Opcode qw(
    opset opset_to_ops opmask_add
    empty_opset full_opset invert_opset verify_opset
    opdesc opcodes opmask define_optag opset_to_hex
);
```

**Comment by rurban** [comment]
Oh, great!

**Comment by rurban** [comment]
Good catch, overly strict XS_VERSION check.
This is something I'll have to fix, but in a seperate ticket.

For now fixed with 71d02b4

**Comment by rurban** [comment]
This was fixed with 5.22.2c
